src/opendr/perception/semantic_segmentation/bisenet/bisenet_learner.py

```diff
@@ -0,0 +1,308 @@
+
+            if verbose:
+                print("Downloading pretrained weights...")
+            file_path = os.path.join(path, "bisenet_camvid.pth")
+
+            if not os.path.exists(file_path):
+                urlretrieve(file_url, file_path)
+
+            file_url = os.path.join(url, "trainedModels", "bisenet_camvid.json")
+
+            if verbose:
+                print("Downloading json file...")
+            file_path = os.path.join(path, "bisenet_camvid.json")
+
+            if not os.path.exists(file_path):
+                urlretrieve(file_url, file_path)
+
+        if mode == "testingImage":
+            file_url = os.path.join(url, "datasets", "testImages", "test1.png")
+            if verbose:
+                print("Downloading a testing image...")
+            file_path = os.path.join(path, "test1.png")
+
+            if not os.path.exists(file_path):
+                urlretrieve(file_url, file_path)
+
+    def save(self, path, verbose=True):
+        if not os.path.isdir(path):
+            os.makedirs(path, exist_ok=True)
+
+        model_name = os.path.basename(path)
+        model_path = os.path.join(path, model_name + ".pth")
+        if verbose:
+            print(model_name)
+        metadata = {"model_paths": [],
+                    "framework": "pytorch",
+                    "format": "pth",
+                    "has_data": False,
+                    "inference_params": {},
+                    "optimized": False,
+                    "optimizer_info": {}}
+        param_filepath = model_name + ".pth"
+        metadata["model_paths"].append(param_filepath)
+        if self.device == 'cpu':
+            torch.save(self.model.state_dict(), model_path)
+        elif self.device == 'cuda':
+            torch.save(self.model.module.state_dict(), model_path)
+        if verbose:
+            print("Model parameters saved.")
+
+        with open(os.path.join(path, model_name + '.json'), 'w', encoding='utf-8') as f:
+            json.dump(metadata, f, ensure_ascii=False, indent=4)
+        if verbose:
+            print("Model metadata saved.")
+
+        return True
+
+    def load(self, path):
+
+        if self.model is None:
+            self.build_model()
+
+        if not os.path.isdir(path):
+            raise FileNotFoundError(f"Could not find directory {path}")
+
+        folder_basename = os.path.basename(path)
+        with open(os.path.join(path, folder_basename + ".json")) as jsonfile:
+            metadata = json.load(jsonfile)
+
+        if self.device == 'cpu':
+            self.model.load_state_dict(torch.load(os.path.join(path, metadata["model_paths"][0])))
+        elif self.device == 'cuda':
+            self.model.module.load_state_dict(torch.load(os.path.join(path, metadata["model_paths"][0])))
+        self.model.eval()
+
+    def reset(self):
+        """This method is not used in this implementation."""
+        return NotImplementedError
```

Comment:
It would be better to change it to
```suggestion
        raise NotImplementedError
```